### PR TITLE
Improve the API documentation and markdown files

### PR DESCRIPTION
**Pull request by ldclakmal**
## Purpose
$subject
Related to: https://github.com/ballerina-platform/ballerina-standard-library/issues/584

## Checklist
- [ ] Linked to an issue
- [ ] Updated the changelog
- [ ] Added tests

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -3,15 +3,19 @@ Ballerina JWT Library
...
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
 
-For more information go to [The JWT Module](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/jwt/latest/jwt/).
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
+
+The Ballerina JWT module facilitates auth providers that is to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```
Comment (MadhukaHarith92) [inline README.md:16]:
```suggestion
The Ballerina JWT module facilitates auth providers that are to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```

File: README.md [hunk 2]
```diff
@@ -3,15 +3,19 @@ Ballerina JWT Library
...
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
 
-For more information go to [The JWT Module](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/jwt/latest/jwt/).
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
```
Comment (MadhukaHarith92) [inline README.md:14]: Is it "claims" or "claim"? Seems it changes along with this description.

File: README.md [hunk 3]
```diff
@@ -97,6 +101,6 @@ All contributors are encouraged to read the [Ballerina Code of Conduct](https://
 
 ## Useful Links
 
-* Discuss about code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
+* Discuss the code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
Comment (MadhukaHarith92) [inline README.md:104]:
```suggestion
* Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```

File: README.md (hunk 2 above)
Comment (ldclakmal) [reply README.md:14]: It is "claims". WDYM by "Seems it changes along with this description."?

File: jwt-ballerina/Package.md [hunk 4]
```diff
@@ -1,9 +1,19 @@
...
+
+The Ballerina JWT module facilitates auth providers that is to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
+
+### Report Issues
+
+To report bugs, request new features, start new discussions, view project boards, etc. please visit Ballerina
+[Standard Library parent repository](https://github.com/ballerina-platform/ballerina-standard-library).
+
+### Useful Links
+- Discuss the code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```
Comment (MadhukaHarith92) [inline jwt-ballerina/Package.md:17]:
```suggestion
- Discuss code changes of the Ballerina project in [ballerina-dev@googlegroups.com](mailto:ballerina-dev@googlegroups.com).
```

File: README.md (hunk 2 above)
Comment (MadhukaHarith92) [reply README.md:14]: Probably I misunderstood it :). Are `multiple` claims in a JWT encoded as a single JSON object? or `multiple` claims in a JWT encoded as `multiple` JSON objects.

Comment (ldclakmal) [reply README.md:14]: Ah yes. "multiple claims in a JWT encoded as a single JSON object" is the correct one. Actually this is the exact definition provided by the RFC :)

File: README.md [hunk 5]
```diff
@@ -3,15 +3,19 @@ Ballerina JWT Library
...
   [![codecov](https://codecov.io/gh/ballerina-platform/module-ballerina-jwt/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerina-jwt)
 
-The JWT library is one of the standard library modules of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
+The JWT library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-It provides a listener JWT authentication provider, which can be used to authenticate the provided credentials against the provided JWT validator configurations, and a client self-signed JWT authentication provider, which can be used to authenticate against an external endpoint with a self-signed JWT issued against the provided JWT issuer configurations. Also, this module provides the functionality related to issuing and validating a JWT.
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (anupama-pathirage) [inline README.md:12]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: jwt-ballerina/Module.md [hunk 6]
```diff
@@ -0,0 +1,54 @@
+## Overview
+
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (anupama-pathirage) [inline jwt-ballerina/Module.md:3]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: jwt-ballerina/Module.md [hunk 7]
```diff
@@ -0,0 +1,54 @@
...
+
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
+
+The Ballerina JWT module facilitates auth providers that is to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```
Comment (anupama-pathirage) [inline jwt-ballerina/Module.md:7]:
```suggestion
The Ballerina JWT module facilitates auth providers that are to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```

File: jwt-ballerina/Package.md [hunk 8]
```diff
@@ -1,9 +1,19 @@
...
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
+
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
+
+The Ballerina JWT module facilitates auth providers that is to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
+
+### Report Issues
+
+To report bugs, request new features, start new discussions, view project boards, etc. please visit Ballerina
```
Comment (anupama-pathirage) [inline jwt-ballerina/Package.md:13]:
```suggestion
To report bugs, request new features, start new discussions, view project boards, etc. please visit the Ballerina
```

File: jwt-ballerina/Package.md [hunk 9]
```diff
@@ -1,9 +1,19 @@
...
+The JWT library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-For information on the operations, which you can perform with this module, see the below **Functions**. For examples on the usage of the operations, see the following.
- * [JWT Issue/Validate Example](https://ballerina.io/learn/by-example/jwt-issue-validate.html)
- * [HTTP Service with JWT Auth Example](https://ballerina.io/learn/by-example/http-service-with-jwt-auth.html)
- * [HTTP Client with Self Signed JWT Auth Example](https://ballerina.io/learn/by-example/http-client-with-self-signed-jwt-auth.html)
- * [HTTP Client with Bearer Token Auth Example](https://ballerina.io/learn/by-example/http-client-with-bearer-token-auth.html)
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515) and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (anupama-pathirage) [inline jwt-ballerina/Package.md:5]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: jwt-ballerina/Package.md [hunk 10]
```diff
@@ -1,9 +1,19 @@
...
+
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
+
+The Ballerina JWT module facilitates auth providers that is to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```
Comment (anupama-pathirage) [inline jwt-ballerina/Package.md:9]:
```suggestion
The Ballerina JWT module facilitates auth providers that are to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
```

File: README.md [hunk 11]
```diff
@@ -3,21 +3,25 @@ Ballerina JWT Library
...
   [![codecov](https://codecov.io/gh/ballerina-platform/module-ballerina-jwt/branch/master/graph/badge.svg)](https://codecov.io/gh/ballerina-platform/module-ballerina-jwt)
 
-The JWT library is one of the standard library modules of the <a target="_blank" href="https://ballerina.io/">Ballerina</a> language.
+The `jwt` library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-It provides a listener JWT authentication provider, which can be used to authenticate the provided credentials against the provided JWT validator configurations, and a client self-signed JWT authentication provider, which can be used to authenticate against an external endpoint with a self-signed JWT issued against the provided JWT issuer configurations. Also, this module provides the functionality related to issuing and validating a JWT.
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (praneesha) [inline README.md:12]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: README.md [hunk 12]
```diff
@@ -3,21 +3,25 @@ Ballerina JWT Library
...
 
-For more information go to [The JWT Module](https://ballerina.io/learn/api-docs/ballerina/#/ballerina/jwt/latest/jwt/).
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
```
Comment (praneesha) [inline README.md:14]:
```suggestion
JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure enabling the claims to be signed digitally or protecting the integrity with a Message Authentication Code(MAC) and/or encrypted.
```

File: README.md [hunk 13]
```diff
@@ -3,21 +3,25 @@ Ballerina JWT Library
...
+
+The Ballerina `jwt` module facilitates auth providers that are to be used by the clients and listeners of different protocol connectors. Also, it provides the APIs for issuing a self-signed JWT and validating a JWT.
+
+For more information go to the [`jwt` module](https://docs.central.ballerina.io/ballerina/jwt/latest).
```
Comment (praneesha) [inline README.md:18]:
```suggestion
For more information, go to the [`jwt` module](https://docs.central.ballerina.io/ballerina/jwt/latest).
```

File: README.md [hunk 14]
```diff
@@ -33,7 +37,7 @@ This repository only contains the source code for the module.
    
         > **Note:** Set the JAVA_HOME environment variable to the path name of the directory into which you installed JDK.
```
Comment (praneesha) [inline README.md:38]:
```suggestion
        > **Note:** Set the `JAVA_HOME` environment variable to the path name of the directory into which you installed JDK.
```

File: README.md [hunk 15]
```diff
@@ -33,7 +37,7 @@ This repository only contains the source code for the module.
...
 
-2. Export Github Personal Access Token (PAT) with read package permissions as follows:
+2. Export GitHub Personal Access Token (PAT) with 'read package' permission as follows:
```
Comment (praneesha) [inline README.md:40]:
```suggestion
2. Export your GitHub Personal Access Token (PAT) with the `read package` permission as follows:
```

File: jwt-ballerina/Module.md [hunk 16]
```diff
@@ -0,0 +1,54 @@
+## Overview
+
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (praneesha) [inline jwt-ballerina/Module.md:3]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: jwt-ballerina/Module.md [hunk 17]
```diff
@@ -0,0 +1,54 @@
...
+
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
```
Comment (praneesha) [inline jwt-ballerina/Module.md:5]:
```suggestion
JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure enabling the claims to be signed digitally or protecting the integrity with a Message Authentication Code(MAC) and/or encrypted.
```

File: jwt-ballerina/Package.md [hunk 18]
```diff
@@ -1,9 +1,18 @@
...
+The `jwt` library is one of the standard library modules of the [Ballerina](https://ballerina.io/) language.
 
-For information on the operations, which you can perform with this module, see the below **Functions**. For examples on the usage of the operations, see the following.
- * [JWT Issue/Validate Example](https://ballerina.io/learn/by-example/jwt-issue-validate.html)
- * [HTTP Service with JWT Auth Example](https://ballerina.io/learn/by-example/http-service-with-jwt-auth.html)
- * [HTTP Client with Self Signed JWT Auth Example](https://ballerina.io/learn/by-example/http-client-with-self-signed-jwt-auth.html)
- * [HTTP Client with Bearer Token Auth Example](https://ballerina.io/learn/by-example/http-client-with-bearer-token-auth.html)
+This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```
Comment (praneesha) [inline jwt-ballerina/Package.md:5]:
```suggestion
This module provides a framework for authentication/authorization with JWTs and generation/validation of JWTs as specified in the [RFC 7519](https://datatracker.ietf.org/doc/html/rfc7519), [RFC 7515](https://datatracker.ietf.org/doc/html/rfc7515), and [RFC 7517](https://datatracker.ietf.org/doc/html/rfc7517).
```

File: jwt-ballerina/Package.md [hunk 19]
```diff
@@ -1,9 +1,18 @@
...
+
+JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be digitally signed or integrity protected with a Message Authentication Code(MAC) and/or encrypted.
```
Comment (praneesha) [inline jwt-ballerina/Package.md:7]:
```suggestion
JSON Web Token (JWT) is a compact, URL-safe means of representing claims to be transferred between two parties. The claims in a JWT are encoded as a JSON object that is used as the payload of a JSON Web Signature (JWS) structure or as the plaintext of a JSON Web Encryption (JWE) structure, enabling the claims to be signed digitally or protecting the integrity with a Message Authentication Code(MAC) and/or encrypted.
```

File: jwt-ballerina/client_self_signed_jwt_auth_provider.bal [hunk 20]
```diff
@@ -43,10 +43,10 @@ public class ClientSelfSignedJwtAuthProvider {
 
     # Issues a self-signed JWT for authentication.
     # ```ballerina
-    # string|auth:Error token = provider.generateToken();
+    # string token = check provider.generateToken();
     # ```
     #
-    # + return - Generated token or else an `auth:Error` if token can't be generated
+    # + return - Generated token or else an `jwt:Error` if an error occurred
```
Comment (praneesha) [inline jwt-ballerina/client_self_signed_jwt_auth_provider.bal:49]:
```suggestion
    # + return - Generated token or else a `jwt:Error` if an error occurred
```

File: jwt-ballerina/jwt_errors.bal [hunk 21]
```diff
@@ -16,8 +16,8 @@
 
 import ballerina/log;
 
-# Represents the JWT error. This will be returned if an error occurred while issuing/validating a JWT or any operation
-# related to JWT.
+# Represents the error type of the module. This will be returned if an error occurred while issuing/validating a JWT
+# or any operation related to JWT auth providers.
 public type Error distinct error;
 
 // Logs and prepares the `error` as an `jwt:Error`.
```
Comment (praneesha) [inline jwt-ballerina/jwt_errors.bal:23]:
```suggestion
// Logs and prepares the `error` as a `jwt:Error`.
```

File: jwt-ballerina/listener_jwt_auth_provider.bal [hunk 22]
```diff
@@ -64,13 +64,13 @@ public class ListenerJwtAuthProvider {
...
-    # Authenticates provided JWT against `jwt:ValidatorConfig`.
+    # Authenticates the provided JWT.
     #```ballerina
-    # boolean|auth:Error result = provider.authenticate("<credential>");
+    # boolean result = check provider.authenticate("<credential>");
     # ```
     #
     # + credential - JWT to be authenticated
-    # + return - `jwt:Payload` if authentication is successful or else an `auth:Error` if JWT validation failed
+    # + return - `jwt:Payload` if authentication is successful or else an `jwt:Error` if an error occurred
```
Comment (praneesha) [inline jwt-ballerina/listener_jwt_auth_provider.bal:73]:
```suggestion
    # + return - `jwt:Payload` if authentication is successful or else a `jwt:Error` if an error occurred
```

File: jwt-ballerina/tests/test_utils.bal [hunk 23]
```diff
@@ -113,7 +113,7 @@ const string JWT4 = "eyJhbGciOiJSUzUxMiIsICJ0eXAiOiJKV1QifQ.eyJpc3MiOiJ3c28yIiwg
 // Build complete error message by evaluating all the inner causes and assert the inclusion.
```
Comment (praneesha) [inline jwt-ballerina/tests/test_utils.bal:113]:
```suggestion
// Builds the complete error message by evaluating all the inner causes and asserts the inclusion.
```

**Pull request merged**